DistroLauncher/OOBE.h

```diff
@@ -0,0 +1,30 @@
+//
+//    Copyright: 2021, Canonical Ltd.
+//  License: GPL-3
+//  This program is free software : you can redistribute itand /or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License.
+//  .
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+//  GNU General Public License for more details.
+//  .
+//  You should have received a copy of the GNU General Public License
+//  along with this program.If not, see < https://www.gnu.org/licenses/>.
+//  .
+//  On Debian systems, the complete text of the GNU General
+//  Public License version 3 can be found in "/usr/share/common-licenses/GPL-3".
+
+#pragma once
+
+namespace DistributionInfo {
+    // OOBE Experience.
+    ULONG OOBE();
+
+    // Generating a YAML string from Windows user and locale information.
+    std::wstring GetPrefillInfoInYaml();
+
+    static const wchar_t* OOBE_NAME = L"ubuntu_wsl_setup";
```

Comment:
Actually, the command line will be `wsl-setup` that will not be in the path on purpose (we want that all interactions to the OOBE is made from the launcher: setup or config, as the launcher will take the reboot/shutdown option that was passed and act accordingly).
It’s going to be installed in `/usr/lib/libexec`, and thus, we need an absolute path, which I think is OK in that case.

You can see it in https://github.com/ubuntu/wsl-setup (if you want, we can spend some time today/tomorrow for a basic presentation of packaging). This package will be available in jammy.
This helper mounts the needed snaps, mock a snap environment and chain it to the snap desktop helper which then (phew!) chain to the actual binary. This is what you pass as an argument: the desktop-installer binary or TUI from subiquity.